A problem occurred evaluating project ':react-native-vision-camera'🐛

### What were you trying to do?

I was running my app without issue with the version 2.10.0, but since the update of 2.11.0 the build failed with an exception.

on the /react-native-vision-camera/android/build.gradle' line: 8
it's looking for the file /react-native-vision-camera/node_modules/react-native/ReactAndroid/gradle.properties 
which doesn't exist (No such file or directory).

I tried to clean the caches but still happening
![Screen Shot 2022-01-03 at 10 48 22](https://user-images.githubusercontent.com/33593109/147912741-8568e364-52c0-4507-b3f9-027d3dcb3d5b.png)



### Reproduceable Code

```tsx
"react-native-vision-camera": "^2.11.0",
```


### What happened instead?

FAILURE: Build failed with an exception.

### Relevant log output

```shell
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/yohanbalassiano/Sites/Deelt/app/deelt/node_modules/react-native-vision-camera/android/build.gradle' line: 8

* What went wrong:
A problem occurred evaluating project ':react-native-vision-camera'.
> /Users/yohanbalassiano/Sites/Deelt/app/deelt/node_modules/react-native-vision-camera/node_modules/react-native/ReactAndroid/gradle.properties (No such file or directory)
```


### Device

OnePlus Nord

### VisionCamera Version

2.11.0

### Additional information

- [ ] I am using Expo
- [X] I have read the [Troubleshooting Guide](https://mrousavy.github.io/react-native-vision-camera/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -2,26 +2,26 @@ import groovy.json.JsonSlurper
 import org.apache.tools.ant.filters.ReplaceTokens
 import java.nio.file.Paths
 
-static def findNodeModulePath(baseDir, packageName) {
+static def findNodeModules(baseDir) {
   def basePath = baseDir.toPath().normalize()
   // Node's module resolution algorithm searches up to the root directory,
   // after which the base path will be null
   while (basePath) {
-    def candidatePath = Paths.get(basePath.toString(), "node_modules", packageName)
+    def candidatePath = Paths.get(basePath.toString(), "node_modules")
     if (candidatePath.toFile().exists()) {
       return candidatePath.toString()
     }
     basePath = basePath.getParent()
   }
-  return null
+  throw new GradleException("VisionCamera: Failed to find node_modules/ path!")
 }
 
-def nodeModulesPath = findNodeModulePath(projectDir, "")
-logger.warn("VisionCamera: node_modules/ found at: ${nodeModulesPath}")
-def reactNative = new File("$nodeModulesPath/react-native")
+def nodeModules = findNodeModules(projectDir)
+logger.warn("VisionCamera: node_modules/ found at: ${nodeModules}")
+def reactNative = new File("$nodeModules/react-native")
 
 def reactProperties = new Properties()
-file("$nodeModulesPath/react-native/ReactAndroid/gradle.properties").withInputStream { reactProperties.load(it) }
+file("$nodeModules/react-native/ReactAndroid/gradle.properties").withInputStream { reactProperties.load(it) }
 def REACT_NATIVE_VERSION = reactProperties.getProperty("VERSION_NAME").split("\\.")[1].toInteger()
 
 def FOR_HERMES = System.getenv("FOR_HERMES") == "True"
@@ -66,7 +66,7 @@ def getExtOrIntegerDefault(name) {
 }
 
 // REA Common/ folder only exists since REA v2.
-def hasReanimated2 = file("${nodeModulesPath}/react-native-reanimated/plugin.js").exists()
+def hasReanimated2 = file("${nodeModules}/react-native-reanimated/plugin.js").exists()
 def disableFrameProcessors = rootProject.ext.has("disableFrameProcessors") ? rootProject.ext.get("disableFrameProcessors").asBoolean() : false
 def ENABLE_FRAME_PROCESSORS = hasReanimated2 && !disableFrameProcessors
 
@@ -96,7 +96,7 @@ android {
           abiFilters 'x86', 'x86_64', 'armeabi-v7a', 'arm64-v8a'
           arguments '-DANDROID_STL=c++_shared',
                   "-DREACT_NATIVE_VERSION=${REACT_NATIVE_VERSION}",
-                  "-DNODE_MODULES_DIR=${nodeModulesPath}",
+                  "-DNODE_MODULES_DIR=${nodeModules}",
                   "-DFOR_HERMES=${FOR_HERMES}"
         }
       }
@@ -226,16 +226,16 @@ dependencies {
     //noinspection GradleDynamicVersion
     extractJNI("com.facebook.fbjni:fbjni:+")
 
-    def rnAAR = fileTree("${nodeModulesPath}/react-native/android").matching({ it.include "**/**/*.aar" }).singleFile
-    def jscAAR = fileTree("${nodeModulesPath}/jsc-android/dist/org/webkit/android-jsc").matching({ it.include "**/**/*.aar" }).singleFile
+    def rnAAR = fileTree("${nodeModules}/react-native/android").matching({ it.include "**/**/*.aar" }).singleFile
+    def jscAAR = fileTree("${nodeModules}/jsc-android/dist/org/webkit/android-jsc").matching({ it.include "**/**/*.aar" }).singleFile
 
-    def inputFile = file("${nodeModulesPath}/react-native/package.json")
+    def inputFile = file("${nodeModules}/react-native/package.json")
     def json = new JsonSlurper().parseText(inputFile.text)
     def reactNativeVersion = json.version as String
     def (major, minor, patch) = reactNativeVersion.tokenize('.')
 
     def jsEngine = FOR_HERMES ? "hermes" : "jsc"
-    def reaAAR = "${nodeModulesPath}/react-native-reanimated/android/react-native-reanimated-${minor}-${jsEngine}.aar"
+    def reaAAR = "${nodeModules}/react-native-reanimated/android/react-native-reanimated-${minor}-${jsEngine}.aar"
 
     extractJNI(files(rnAAR, jscAAR, reaAAR))
   }
@@ -262,7 +262,7 @@ if (ENABLE_FRAME_PROCESSORS) {
 
   def downloadsDir = new File("$buildDir/downloads")
   def thirdPartyNdkDir = new File("$buildDir/third-party-ndk")
-  def thirdPartyVersionsFile = new File("${nodeModulesPath}/react-native/ReactAndroid/gradle.properties")
+  def thirdPartyVersionsFile = new File("${nodeModules}/react-native/ReactAndroid/gradle.properties")
   def thirdPartyVersions = new Properties()
   thirdPartyVersions.load(new FileInputStream(thirdPartyVersionsFile))
 
@@ -394,8 +394,8 @@ if (ENABLE_FRAME_PROCESSORS) {
 
   task prepareHermes() {
     doLast {
-      def hermesPackagePath = findNodeModulePath(projectDir, "hermes-engine")
-      if (!hermesPackagePath) {
+      def hermesPackagePath = file("${nodeModules}/hermes-engine")
+      if (!hermesPackagePath.exists()) {
         throw new GradleScriptException("Could not find the hermes-engine npm package", null)
       }
 
@@ -418,8 +418,8 @@ if (ENABLE_FRAME_PROCESSORS) {
 
   task prepareJSC {
     doLast {
-      def jscPackagePath = findNodeModulePath(projectDir, "jsc-android")
-      if (!jscPackagePath) {
+      def jscPackagePath = file("${nodeModules}/jsc-android")
+      if (!jscPackagePath.exists()) {
         throw new GradleScriptException("Could not find the jsc-android npm package", null)
       }
 

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -16,7 +16,7 @@ static def findNodeModulePath(baseDir, packageName) {
   return null
 }
 
-def nodeModulesPath = findNodeModulePath(projectDir, null)
+def nodeModulesPath = findNodeModulePath(projectDir, "")
 def reactNative = new File("$nodeModulesPath/react-native")
 
 def reactProperties = new Properties()

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -2,41 +2,6 @@ import groovy.json.JsonSlurper
 import org.apache.tools.ant.filters.ReplaceTokens
 import java.nio.file.Paths
 
-def nodeModulesPath = "$projectDir/../node_modules"
-def reactNative = new File("$nodeModulesPath/react-native")
-
-def reactProperties = new Properties()
-file("$nodeModulesPath/react-native/ReactAndroid/gradle.properties").withInputStream { reactProperties.load(it) }
-def REACT_NATIVE_VERSION = reactProperties.getProperty("VERSION_NAME").split("\\.")[1].toInteger()
-
-def FOR_HERMES = System.getenv("FOR_HERMES") == "True"
-rootProject.getSubprojects().forEach({project ->
-  if (project.plugins.hasPlugin("com.android.application")) {
-    FOR_HERMES = project.ext.react.enableHermes
-  }
-})
-
-/**
- * Finds the path of the installed npm package with the given name using Node's
- * module resolution algorithm, which searches "node_modules" directories up to
- * the file system root. This handles various cases, including:
- *
- *   - Working in the open-source RN repo:
- *       Gradle: /path/to/react-native/ReactAndroid
- *       Node module: /path/to/react-native/node_modules/[package]
- *
- *   - Installing RN as a dependency of an app and searching for hoisted
- *     dependencies:
- *       Gradle: /path/to/app/node_modules/react-native/ReactAndroid
- *       Node module: /path/to/app/node_modules/[package]
- *
- *   - Working in a larger repo (e.g., Facebook) that contains RN:
- *       Gradle: /path/to/repo/path/to/react-native/ReactAndroid
- *       Node module: /path/to/repo/node_modules/[package]
- *
- * The search begins at the given base directory (a File object). The returned
- * path is a string.
- */
 static def findNodeModulePath(baseDir, packageName) {
   def basePath = baseDir.toPath().normalize()
   // Node's module resolution algorithm searches up to the root directory,
@@ -50,6 +15,20 @@ static def findNodeModulePath(baseDir, packageName) {
   }
   return null
 }
+
+def nodeModulesPath = findNodeModulePath(projectDir, null)
+def reactNative = new File("$nodeModulesPath/react-native")
+
+def reactProperties = new Properties()
+file("$nodeModulesPath/react-native/ReactAndroid/gradle.properties").withInputStream { reactProperties.load(it) }
+def REACT_NATIVE_VERSION = reactProperties.getProperty("VERSION_NAME").split("\\.")[1].toInteger()
+
+def FOR_HERMES = System.getenv("FOR_HERMES") == "True"
+rootProject.getSubprojects().forEach({project ->
+  if (project.plugins.hasPlugin("com.android.application")) {
+    FOR_HERMES = project.ext.react.enableHermes
+  }
+})
 
 buildscript {
   // Buildscript is evaluated before everything else so we can't use getExtOrDefault

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -17,6 +17,7 @@ static def findNodeModulePath(baseDir, packageName) {
 }
 
 def nodeModulesPath = findNodeModulePath(projectDir, "")
+logger.warn("VisionCamera: node_modules/ found at: ${nodeModulesPath}")
 def reactNative = new File("$nodeModulesPath/react-native")
 
 def reactProperties = new Properties()

